server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1835,6 +1847,26 @@ private Volume orchestrateDetachVolumeFromVM(long vmId, long volumeId) {
         }
     }
 
+    public void updateMissingRootDiskController(final VMInstanceVO vm, final String rootVolChainInfo) {
+        if (vm == null || !vm.getType().equals(VirtualMachine.Type.User) || Strings.isNullOrEmpty(rootVolChainInfo)) {
```

Comment:
the order of evalution will be left to right -- the first case is vm == null; if this qualifies the if statement will branch into executing code; otherwise vm.getType() code will be executed (i.e. vm is not null); now here I think you're suggesting that perhaps vm.getType() may return null in which case we may have a NPE (unlikely based on db contraints (the type column should not be null), but I'll modify as suggested.